code/modules/power/supermatter/supermatter.dm

```diff
@@ -678,6 +678,24 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 		if(produces_gas)
 			env.merge(removed)
 			air_update_turf()
+	if(takes_damage)
+		for(var/turf_to_check in RANGE_TURFS(1, loc))
+			if(isspaceturf(turf_to_check) && istype(loc))
+				var/is_open =  TRUE
+				for(var/obj/object_to_check in turf_to_check)
+					if(object_to_check.density == TRUE || object_to_check.CanAtmosPass == ATMOS_PASS_NO)
```

Comment:
If all you want to check is if the turf itself passes atmos or not, do if(atmos_adjacent_turfs.len). If that list is empty, the turf doesn't share with anything, and can be assumed to be covered.